src/js/hooks/useProductFilters.jsx

```diff
@@ -0,0 +1,137 @@
+import { useEffect, useState } from 'react';
+
+import queryString from 'query-string';
+import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import filterFields from 'components/products/FilterFields';
+import apiClient from 'utils/apiClient';
+import { getParamList, transformFilterParams } from 'utils/list-utils';
+
+const useProductFilters = () => {
+  const [filterParams, setFilterParams] = useState({});
+  const [defaultFilterValues, setDefaultFilterValues] = useState({});
+  const [categories, setCategories] = useState([]);
+  const [catalogs, setCatalogs] = useState([]);
+  const [tags, setTags] = useState([]);
+  const [filtersInitialized, setFiltersInitialized] = useState(false);
+
+  const history = useHistory();
+  const { currentLocation } = useSelector(state => ({
+    currentLocation: state.session.currentLocation,
+  }));
+  const fetchProductsCategories = async () => {
+    const response = await apiClient.get('/openboxes/api/categoryOptions');
+    return response.data.data;
+  };
+
+  const fetchProductsCatalogs = async () => {
+    const response = await apiClient.get('/openboxes/api/catalogOptions');
+    return response.data.data;
+  };
+
+  const fetchProductsTags = async () => {
+    const response = await apiClient.get('/openboxes/api/tagOptions');
+    return response.data.data;
+  };
```

Comment:
I'm wondering if those three functions have to be inside the component and re-render each time? I know it's not you, who pulled it inside the component, but maybe it would be a good idea to move those functions to `utils/option-utils`, as those functions could be potentially useful in the future.